core/src/gl/framebuffer.cpp

```diff
@@ -72,6 +72,21 @@ GLuint FrameBuffer::readAt(float _normalizedX, float _normalizedY) const {
     return pixel;
 }
 
+FrameBuffer::PixelRect FrameBuffer::readRect(float _normalizedX, float _normalizedY, float _normalizedW, float _normalizedH) const {
+
+    PixelRect rect;
+    rect.left = floorf(_normalizedX * m_width);
+    rect.bottom = floorf(_normalizedY * m_height);
+    rect.width = ceilf(_normalizedW * m_width);
+    rect.height = ceilf(_normalizedH * m_height);
+
+    rect.pixels.resize(rect.width * rect.height);
+
+    GL::readPixels(rect.left, rect.bottom, rect.width, rect.height, GL_RGBA, GL_UNSIGNED_BYTE, rect.pixels.data());
```

Comment:
Since we have a radius now it is possible that we make viewport read out of the window bounds. Do you know the behavior in that case? Definitely not an error by reading the docs, but it would be worth checking whether this is a potential memory corruption.